x-pack/plugins/encrypted_saved_objects/server/crypto/encrypted_saved_objects_service.ts

```diff
@@ -152,6 +153,16 @@ export class EncryptedSavedObjectsService {
     return this.typeDefinitions.has(type);
   }
 
+  /**
+   * Checks whether the ESO type has explicitly opted out of enforcing random IDs.
+   * @param type Saved object type.
+   * @returns boolean - true unless explicitly opted out by setting enforceRandomId to false
+   */
+  public shouldEnforceRandomId(type: string) {
+    const typeDefinition = this.typeDefinitions.get(type);
+    return typeDefinition?.enforceRandomId !== false;
```

Comment:
>[!TIP]
> Optional and not that important, but I’m wondering if we should throw an error here if type isn’t registered (`isEncryptableType(type) !== true`)? It’s unlikely to happen in practice since we expect consumers of this API to call `isEncryptableType` first, but it still feels a bit odd that this function would return `true` for non-ESOs if someone mistakenly uses it in this context.

Reply:
@azasypkin  That's a good point! I think i'll address it in a follow-up PR.  

I was also thinking about the scenario where a type explicitly opts out but no ID is provided during creation. In it's current implementation, a random ID will be created but I was wondering if it should behave differently?

Reply:
>@azasypkin That's a good point! I think i'll address it in a follow-up PR.

Thanks!

> I was also thinking about the scenario where a type explicitly opts out but no ID is provided during creation. In it's current implementation, a random ID will be created but I was wondering if it should behave differently?

Hmm, I think the current behavior is correct and works the same for both ESO and non-ESO saved objects. I interpret `enforceRandomId` as "prevent consumers from explicitly specifying non-random IDs, but keep the default behavior if they don’t specify IDs".

Reply:
++ Yeah, if an type has opted out of enforcing random IDs but an ID is not provided during creation, a random ID should be generated.